### DISCUSSION TITLE
send help! trying to install on slackware !!! (long post)

### DISCUSSION THREAD

**Discussion by hacknorris-code** [misskey-dev/misskey#9151] [General]
os info : 
```
                  :::::::                      root@hacknorris.hacknorris.com 
            :::::::::::::::::::                ------------------------------ 
         :::::::::::::::::::::::::             OS: Slackware 15.0 x86_64 x86_64 
       ::::::::cllcccccllllllll::::::          Host: Dynabook Inc. DBIIP300 
    :::::::::lc               dc:::::::        Kernel: 5.15.19 
   ::::::::cl   clllccllll    oc:::::::::      Uptime: 1 hour, 1 min 
  :::::::::o   lc::::::::co   oc::::::::::     Packages: 1598 (pkgtool) 
 ::::::::::o    cccclc:::::clcc::::::::::::    Shell: bash 5.1.16 
 :::::::::::lc        cclccclc:::::::::::::    Resolution: 1920x1080 
::::::::::::::lcclcc          lc::::::::::::   DE: Plasma 5.23.5 
::::::::::cclcc:::::lccclc     oc:::::::::::   WM: KWin 
::::::::::o    l::::::::::l    lc:::::::::::   Theme: Breeze Light [Plasma], Breeze [GTK2/3] 
 :::::cll:o     clcllcccll     o:::::::::::    Icons: [Plasma], breeze-dark [GTK2/3] 
 :::::occ:o                  clc:::::::::::    Terminal: konsole 
  ::::ocl:ccslclccclclccclclc:::::::::::::     CPU: Intel i3-8130U (4) @ 3.400GHz 
   :::oclcccccccccccccllllllllllllll:::::      GPU: Intel UHD Graphics 620 
    ::lcc1lcccccccccccccccccccccccco::::       Memory: 3462MiB / 7859MiB 
      ::::::::::::::::::::::::::::::::
        ::::::::::::::::::::::::::::                                   
           ::::::::::::::::::::::                                      
                ::::::::::::

```

things done : 

- [x] installed dependencies properly
- [x] downloaded misskey folder properly
- [x] setted up system users (unsure if ever properly cause adduser doesnt support flags and useradd doesnt have nopassword flag so im having probably a problem)
- [x] yarn install successful

---

thing NOT done and trying to do:

- [x] setup postgres

---

why i cant do it?

error log:

bug from stackoverflow answer (you didnt mentioned it on page):

```
root@hacknorris:/# su postgres -c "pg_ctl -D /var/lib/pgsql/14/data -l logfile start"
waiting for server to start..../bin/sh: line 1: logfile: Permission denied
 stopped waiting
pg_ctl: could not start server
Examine the log output.
```

another stackoverflow thing which i got... from slackware matrix folks... o.O

```
root@hacknorris:/# su postgres -c "initdb -D /var/lib/pgsql/14/data --locale=en_US.UTF-8 -A md5 -W"
The files belonging to this database system will be owned by user "postgres".
This user must also own the server process.

The database cluster will be initialized with locale "en_US.UTF-8".
The default database encoding has accordingly been set to "UTF8".
The default text search configuration will be set to "english".

Data page checksums are disabled.

Enter new superuser password: 
Enter it again: 

initdb: error: directory "/var/lib/pgsql/14/data" exists but is not empty
If you want to create a new database system, either remove or empty
the directory "/var/lib/pgsql/14/data" or run initdb
with an argument other than "/var/lib/pgsql/14/data".

```

your bug (idk where a bug, either in useradd/adduser or postgres itself, first try was my root password, second nothing, later left laptop with rage):

```
misskey@hacknorris:~$ sudo -u postgres psql

We trust you have received the usual lecture from the local System
Administrator. It usually boils down to these three things:

#1) Respect the privacy of others.
#2) Think before you type.
#3) With great power comes great responsibility.

Password:
I feel much better now.
Password:
I've seen penguins that can type better than that.
Password:
sudo: timed out reading password
sudo: 2 incorrect password attempts
```

my try to reinterprete tutorial made this : 

```
root@hacknorris:~# su postgres
postgres@hacknorris:/root$ psql
could not change directory to "/root": Permission denied
psql: error: connection to server on socket "/tmp/.s.PGSQL.5432" failed: No such file or directory
        Is the server running locally and accepting connections on that socket?
```

anyone ?

yes, i also would to have you official on matrix/irc/other place, not only on propertiary plataforms like here or discord (thank god i didnt yet shutted this github :/ )

**Comment by ForestJohnson** [comment]
Looks like postgres wont start because of permissions.   

Sorry about the hostility you faced in #selfhosted:matrix.org,  those folks are very impatient and elitist.  You will get more help from smaller, more tight knit communities where empathy and treating people like people is part of the norms / rules. 

Ultimately it looks like either your postrgres installation is messed up,  or the postgres package for slackware is messed up.    Normally when software like this is installed, the installer ( slackware package in this case )  is supposed to set up all the permissions of various different folders correctly so that the software can run.  However in this case it appears that this did not happen.  

If you wish to see the "smoking gun"  you may be able to use syscall-level debugging tools like [strace](https://strace.io/) to instrument the [filesystem related system calls that postgres application process(es)](https://sequentialread.com/what-is-a-process/#overviewofprocesses) are making to the OS. Then you can see exactly which one fails.
However that would be a whole nother thing to install and learn about, if  you've never used it before, it's probably a pretty steep learning curve. 

We already know a little bit about it: `waiting for server to start..../bin/sh: line 1: logfile: Permission denied` 

 so it sounds like some start script (presumably running as the postgres user) is trying to open a file named logfile, however the postgres user does not have permission to open this file. 
 
 Here is a result I found on google search when I searched for that error message:  https://www.postgresql.org/message-id/44A82CC5.8030300%40gmail.com
 
 It sounds like perhaps this start script is writing a file into the current directory?  That sounds bizzarre but it does sort of jive with what you are seeing. So maybe you can try making a directory thats owned by postgres,   `cd` into that directory, and then try to start postgres.
 
If that doesn't work, see if you can locate the start script itself and edit it. Try adding debug `echo` or `printf` statements to it.  See if you can run `pwd` in there to output which directory its running in.  Then you will know which directory to `chown postgres:postgres` 
 
 Might also be worth it to search the process list for postgres to see if any other processes named postgres are already running.  Kill any postgres related processes before trying to start it again.

**Reply by hacknorris-code** [reply]
actually i already fixed that... was quite simple lol but anyway - now the only thing i want is to nuke my isp but this is not a topic for here anyway =]
